sdks/python/hatchet_sdk/worker/action_listener_process.py

```diff
@@ -291,6 +291,9 @@ async def cleanup(self) -> None:
         self.event_queue.put(STOP_LOOP)
 
     async def exit_gracefully(self) -> None:
+        if self.listener:
+            self.listener.stop_signal = True
```

Comment:
@grutt would like a sanity check here - I was thinking this should forcibly stop assignment to the worker (even before the pause processes). lmk if this is wrong